pkg/controlplane/apiserver/server.go

```diff
@@ -146,6 +148,25 @@ func (c completedConfig) New(name string, delegationTarget genericapiserver.Dele
 		return nil, fmt.Errorf("failed to get listener address: %w", err)
 	}
 
+	if utilfeature.DefaultFeatureGate.Enabled(apiserverfeatures.CoordinatedLeaderElection) {
+		s.GenericAPIServer.AddPostStartHookOrDie("start-kube-apiserver-coordinated-leader-election-controller", func(hookContext genericapiserver.PostStartHookContext) error {
+			go leaderelection.RunWithLeaderElection(hookContext, s.GenericAPIServer.LoopbackClientConfig, func() (func(ctx context.Context, workers int), error) {
+				// Coordinated Leader Election requires its own SharedInformerFactory because the leader
+				// may be acquired any time. apiserver informers will have started already and cannot be used.
```

Comment:
why is that? You can start informers at any time.

Reply:
right, but we can't start it multiple times? One apiserver instance will win the lock and be the leader. If it shuts down and another instance takes over, the default apiserver informers will have already been registered/started. At least it seems to be a convention for any leader elected component to start informers AFTER confirming leadership.